README
NAME
    Data::Semantic::Net - Semantic data classes for net-related data

VERSION
    version 1.101750

DESCRIPTION
    This class is a base class for net-related semantic data classes. The
    following classes are available:

    IPv4
        See Data::Semantic::Net::IPAddress::IPv4.

    IPv6
        See Data::Semantic::Net::IPAddress::IPv6.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Data-Semantic-Net/>.

    The development version lives at
    <http://github.com/hanekomu/Data-Semantic-Net/>. Instead of sending
    patches, please fork this project using the standard git and github
    infrastructure.

AUTHOR
      Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2007 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Data-Semantic-Net

1.101750  2010-06-24 12:49:16 Europe/Vienna
    - improved checking whether an IPv6 address is internal
    - added IPv6 test data, also normalization tests

1.101730  2010-06-22 14:27:43 Europe/Vienna
    - Data::Semantic::Net::IPAddress::TestData::IPv4: specifically add
      'forbid_internal => 0' where appropriate now that the default is 1

1.101610  2010-06-10 21:55:48 Europe/Vienna
    - added normalize() to Data::Semantic::Net::IPAddress::IPv4, also test data

1.100850  2010-03-26 14:09:59 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.04  Thu Jul 25 15:27:43 CEST 2008 (Marcel Gruenauer <marcel@cpan.org>)
    - Changed dist style and Changes back to standard. People didn't like it -
      the nail that sticks out gets hammered down.
    - Added standard test files; this will also help with CPANTS.

0.03  Thu, 29 May 2008 12:03:38 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - set the version to 0.03
    - moved generated documentation of interited methods to after documentation
      of this class' methods so the manpage is more readable
    - .shipit: fixed svk.tagpattern

0.02  Thu, 22 Nov 2007 13:08:54 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - set the version to 0.02
    - Makefile.PL: now requires Test::CompanionClasses 0.03; earlier versions
      had a bug in make_real_object()

0.01  Wed, 21 Nov 2007 18:22:09 -0000 (Marcel Gruenauer <marcel@cpan.org>)
    - original version
    - Makefile.PL: fixed requirements
    - Data::Semantic::Net::IPAddress::IPv4_TEST: instead of defining values
      itself, it inherits also from
      Data::Semantic::Net::IPAddress::TestData::IPv4
    - Data::Semantic::Net::IPAddress: add boolean accessor forbid_internal(),
      is_valid_normalized_value() uses it
    - updated MANIFEST
    - added lib/Data/Semantic/Net/IPAddress/IPv6_TEST.pm
    - added lib/Data/Semantic/Net/IPAddress/TestData
    - getting rid of the distinction between well-formedness and validity.
      Only validity remains
    - Data::Semantic::Base is no more; use Data::Semantic instead
    - added documentation
    - updated t/perlcriticrc


inc/Test/CompanionClasses.pm
#line 1
package Test::CompanionClasses;

use warnings;
use strict;
use Test::CompanionClasses::Engine;
use Getopt::Long;


use base 'Exporter';


our $VERSION = '0.03';


our @EXPORT = ('run_tests');


sub run_tests {
    my $exact;
    GetOptions(exact => \$exact) or
        die "usage: $0 [ --exact ] filter...\n";

    Test::CompanionClasses::Engine->new->run_tests(
        exact     => $exact,
        filter    => [ @main::ARGV ],
        # inherited => [ $inherited_spec ],
    );
}



__END__

{% USE p = PodGenerated %}

#line 78



lib/Data/Semantic/Net/IPAddress/IPv4_TEST.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net::IPAddress::IPv4_TEST;
BEGIN {
  $Data::Semantic::Net::IPAddress::IPv4_TEST::VERSION = '1.101750';
}

use Test::More;

# ABSTRACT: Test companion class for the IPv4 address semantic data class
use parent qw(
  Data::Semantic::Test
  Data::Semantic::Net::IPAddress::TestData::IPv4
);

1;


__END__
=pod

=head1 NAME

Data::Semantic::Net::IPAddress::IPv4_TEST - Test companion class for the IPv4 address semantic data class

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

Test companion class for L<Data::Semantic::Net::IPAddress::IPv4>. Gets its
test data from L<Data::Semantic::Net::IPAddress::TestData::IPv4>.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Data/Semantic/Net/IPAddress/IPv6_TEST.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net::IPAddress::IPv6_TEST;
BEGIN {
  $Data::Semantic::Net::IPAddress::IPv6_TEST::VERSION = '1.101750';
}
# ABSTRACT: Test companion class for the IPv6 address semantic data class
use parent qw(
  Data::Semantic::Test
  Data::Semantic::Net::IPAddress::TestData::IPv6
);
1;


__END__
=pod

=head1 NAME

Data::Semantic::Net::IPAddress::IPv6_TEST - Test companion class for the IPv6 address semantic data class

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

Test companion class for L<Data::Semantic::Net::IPAddress::IPv6>. Gets its
test data from L<Data::Semantic::Net::IPAddress::TestData::IPv6>.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Data/Semantic/Net.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net;
BEGIN {
  $Data::Semantic::Net::VERSION = '1.101750';
}
# ABSTRACT: Semantic data classes for net-related data
use parent qw(Data::Semantic);
1;


__END__
=pod

=head1 NAME

Data::Semantic::Net - Semantic data classes for net-related data

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

This class is a base class for net-related semantic data classes. The
following classes are available:

=over 4

=item IPv4

See L<Data::Semantic::Net::IPAddress::IPv4>.

=item IPv6

See L<Data::Semantic::Net::IPAddress::IPv6>.

=back

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Data/Semantic/Net/IPAddress.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net::IPAddress;
BEGIN {
  $Data::Semantic::Net::IPAddress::VERSION = '1.101750';
}
# ABSTRACT: Base class for IP address semantic data classes
use parent qw(Data::Semantic::Net);
__PACKAGE__
    ->mk_abstract_accessors(qw(is_internal))
    ->mk_boolean_accessors(qw(forbid_internal));

sub is_valid_normalized_value {
    my ($self, $value) = @_;
    return unless $self->SUPER::is_valid_normalized_value($value);
    $self->forbid_internal ? !$self->is_internal($value) : 1;
}

1;


__END__
=pod

=head1 NAME

Data::Semantic::Net::IPAddress - Base class for IP address semantic data classes

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

This class is a base class for semantic data objects representing IP addresses
- IPv4 and IPv6.

=head1 METHODS

=head2 is_valid_normalized_value

FIXME

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Data/Semantic/Net/IPAddress/TestData/IPv4.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net::IPAddress::TestData::IPv4;
BEGIN {
  $Data::Semantic::Net::IPAddress::TestData::IPv4::VERSION = '1.101750';
}
# ABSTRACT: Test data class for the IPv4 address semantic data class
use constant TESTDATA => (
    {   args  => { forbid_internal => 0 },
        valid => [
            qw(
              127.0.0.1
              131.130.249.31
              10.0.1.3
              192.168.1.3
              )
        ],
        invalid => [
            qw(
              127.0
              )
        ],
        normalize => {
            '213.160.065.064' => '213.160.65.64',
            '213.065.064'     => undef,
        },
    },
    {   args  => { forbid_internal => 1 },
        valid => [
            qw(
              131.130.249.31
              )
        ],
        invalid => [
            qw(
              127.0
              127.0.0.1
              10.0.1.3
              192.168.1.3
              )
        ],
    },
);
1;


__END__
=pod

=head1 NAME

Data::Semantic::Net::IPAddress::TestData::IPv4 - Test data class for the IPv4 address semantic data class

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

Defines test data for L<Data::Semantic::Net::IPAddress::IPv4_TEST>, but it is
also used in the corresponding value and domain classes, i.e.,
L<Class::Value::Net::IPAddress::IPv4_TEST> and
L<Data::Domain::Net::IPAddress::IPv4_TEST>.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



lib/Data/Semantic/Net/IPAddress/TestData/IPv6.pm
use 5.008;
use strict;
use warnings;

package Data::Semantic::Net::IPAddress::TestData::IPv6;
BEGIN {
  $Data::Semantic::Net::IPAddress::TestData::IPv6::VERSION = '1.101750';
}

# ABSTRACT: Test data class for the IPv4 address semantic data class
use constant TESTDATA => (
    {   args  => { forbid_internal => 0 },
        valid => [
            qw(
              ::
              ::0
              001::
              ::1
              1::
              1000::
              ::131:130:1:12
              2001:0db8::
              2001:0db8:0000:0000:0000:0000:1428:57ab
              2001:0db8:0000:0000:0000::1428:57ab
              2001:0db8:0:0:0:0:1428:57ab
              2001:0db8:0:0::1428:57ab
              2001:0db8::1:0
              2001:0db8::1:0:0
              2001:0db8::1:0:0:0
              2001:0db8::1:0:0:0:0
              2001:0db8:1:0:0:0:0:0
              2001:0db8::1428:57ab
              2001::1
              2001::1:0
              2001:10::
              2001::1:0:0
              2001::1:0:0:0
              2001::1:0:0:0:0
              2001::1:0:0:0:0:0
              2001:10::dead
              2001:db8::1
              2001:db8::1428:57ab
              2001:dead:dead::10:5
              2345::1:0:0:0:2:0
              2345::1:0:0:2:0
              2345::1:0:0:2:0:0
              2345::1:0:2:0
              2345::1:0:2:0:0
              2345::1:0:2:0:0:0
              3000::1
              4000::1
              )
        ],
        invalid => [
            qw(
              0127.0000.01.000000004
              10.0.1.3
              127.0
              127.0.0.1
              127.0.0.1
              131.130.249.31
              192.168.1.3
              2001:00db8::
              2001:0db8::1:0:0:0:0:0:0
              2001:0db8::1:0:0:0:0:0:0:0
              2345::1::2
              )
        ],
        normalize => {
            '::'                     => '::',
            '::0'                    => '::',
            '001::'                  => '1::',
            '0127.0000.01.000000004' => '127.0.1.4',
            '1000::'                 => '1000::',
            '::1'                    => '::1',
            '1::'                    => '1::',
            '2001:0db8:1:0:0:0:0:0'  => '2001:db8:1::',
            '2001:0db8::1:0:0:0:0'   => '2001:db8:0:1::',
            '2001:0db8::1:0:0:0'     => '2001:db8:0:0:1::',
            '2001:0db8::1:0:0'       => '2001:db8::1:0:0',
            '2001:0db8::1:0'         => '2001:db8::1:0',
            '2001:0db8::'            => '2001:db8::',
            '2001::1:0:0:0:0:0'      => '2001:0:1::',
            '2001::1:0:0:0:0'        => '2001:0:0:1::',
            '2001::1:0:0:0'          => '2001::1:0:0:0',
            '2001::1:0:0'            => '2001::1:0:0',
            '2001::1:0'              => '2001::1:0',
            '2001:10::'              => '2001:10::',
            '2001:10::dead'          => '2001:10::dead',
            '2001::1'                => '2001::1',
            '2001:db8::1'            => '2001:db8::1',
            '2001:dead:dead::10:5'   => '2001:dead:dead::10:5',
            '2345::1:0:0:0:2:0'      => '2345:0:1::2:0',
            '2345::1:0:0:2:0:0'      => '2345:0:1::2:0:0',
            '2345::1:0:0:2:0'        => '2345::1:0:0:2:0',
            '2345::1:0:2:0:0:0'      => '2345:0:1:0:2::',
            '2345::1:0:2:0:0'        => '2345::1:0:2:0:0',
            '2345::1:0:2:0'          => '2345::1:0:2:0',
            '3000::1'                => '3000::1',
            '4000::1'                => '4000::1',
        },
    },
    {   args  => { forbid_internal => 1 },
        valid => [
            qw(
              2001::1
              2001::1:0
              2001:10::
              2001::1:0:0
              2001::1:0:0:0
              2001::1:0:0:0:0
              2001::1:0:0:0:0:0
              2001:10::dead
              2001:dead:dead::10:5
              2345::1:0:0:0:2:0
              2345::1:0:0:2:0
              2345::1:0:0:2:0:0
              2345::1:0:2:0
              2345::1:0:2:0:0
              2345::1:0:2:0:0:0
              3000::1
              )
        ],
        invalid => [
            qw(
              ::
              ::0
              001::
              0127.0000.01.000000004
              ::1
              1::
              1000::
              10.0.1.3
              127.0
              127.0.0.1
              127.0.0.1
              ::131:130:1:12
              131.130.249.31
              192.168.1.3
              2001:00db8::
              2001:0db8::
              2001:0db8:0000:0000:0000:0000:1428:57ab
              2001:0db8:0000:0000:0000::1428:57ab
              2001:0db8:0:0:0:0:1428:57ab
              2001:0db8:0:0::1428:57ab
              2001:0db8::1:0
              2001:0db8::1:0:0
              2001:0db8::1:0:0:0
              2001:0db8::1:0:0:0:0
              2001:0db8:1:0:0:0:0:0
              2001:0db8::1:0:0:0:0:0:0
              2001:0db8::1:0:0:0:0:0:0:0
              2001:0db8::1428:57ab
              2001:db8::1
              2001:db8::1428:57ab
              2345::1::2
              4000::1
              )
        ],
    },
);
1;


__END__
=pod

=head1 NAME

Data::Semantic::Net::IPAddress::TestData::IPv6 - Test data class for the IPv4 address semantic data class

=head1 VERSION

version 1.101750

=head1 DESCRIPTION

Defines test data for L<Data::Semantic::Net::IPAddress::IPv6_TEST>, but it is
also used in the corresponding value and domain classes, i.e.,
L<Class::Value::Net::IPAddress::IPv6_TEST> and
L<Data::Domain::Net::IPAddress::IPv6_TEST>.

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

No bugs have been reported.

Please report any bugs or feature requests through the web interface at
L<http://rt.cpan.org>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see
L<http://search.cpan.org/dist/Data-Semantic-Net/>.

The development version lives at
L<http://github.com/hanekomu/Data-Semantic-Net/>.
Instead of sending patches, please fork this project using the standard git
and github infrastructure.

=head1 AUTHOR

  Marcel Gruenauer <marcel@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2007 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

 BEGIN { require 5.008; } 

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'Semantic data classes for net-related data',
  'AUTHOR' => 'Marcel Gruenauer <marcel@cpan.org>',
  'BUILD_REQUIRES' => {
    'Carp' => '0',
    'English' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'Scalar::Util' => '0',
    'Test::CompanionClasses' => '0',
    'Test::More' => '0.94'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Data-Semantic-Net',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Data::Semantic::Net',
  'PREREQ_PM' => {
    'Data::Semantic' => '0',
    'Data::Semantic::Test' => '0',
    'Net::IP' => '0',
    'Test::More' => '0',
    'constant' => '0',
    'parent' => '0'
  },
  'VERSION' => '1.101750',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



